File: TF_Tooling_Web/src/Tracking/TrackingManager.ts
import { ConnectionManager } from '../Connection/ConnectionManager';
import { TrackingStateResponse } from '../Connection/TouchFreeServiceTypes';
import { TrackingState } from './TrackingTypes';

/**
 * This class provides methods for getting and setting the settings of the tracking software.
 * @public
 */
export class TrackingManager {
    
    /**
     * Request a {@link TrackingStateResponse} representing the current state of the tracking software
     * @remarks
     * Use {@link ConvertResponseToState} on the response to get TrackingState in a more helpful form
     * @param _callback Callback to call with {@link TrackingStateResponse}
     */
    public static RequestTrackingState(_callback: (detail: TrackingStateResponse) => void) {
        if (!_callback) {
            console.error('Config file state request failed. This call requires a callback.');
            return;
        }

        ConnectionManager.serviceConnection()?.RequestTrackingState(_callback);
    }

    /**
     * Requests a modification to the tracking software's settings.
     * @param _state State to request. Options not provided within the object will not be modified.
     * @param _callback Optional callback if you require confirmation that settings were changed correctly.
     */
    public static RequestTrackingChange(
        _state: Partial<TrackingState>,
        _callback: ((detail: TrackingStateResponse) => void) | null = null
    ): void {
        ConnectionManager.serviceConnection()?.RequestTrackingChange(_state, _callback);
    }

    /**
     * Converts a {@link TrackingStateResponse} to a `Partial<TrackingState>` to make the response easier to consume.
     * @param _response Response to convert
     * @returns Converted Partial {@link TrackingState}
     */
    public static ConvertResponseToState(_response: TrackingStateResponse): Partial<TrackingState> {
        const response: Partial<TrackingState> = {};

        if (_response.mask !== undefined && _response.mask !== null) {
            response.mask = _response.mask.content;
        }

        if (_response.cameraReversed !== undefined && _response.cameraReversed !== null) {
            response.cameraReversed = _response.cameraReversed.content;
        }

        if (_response.allowImages !== undefined && _response.allowImages !== null) {
            response.allowImages = _response.allowImages.content;
        }

        if (_response.analyticsEnabled !== undefined && _response.analyticsEnabled !== null) {
            response.analyticsEnabled = _response.analyticsEnabled.content;
        }

        return response;
    }
}

File: TF_Settings_and_Tooling_Unity/Assets/TouchFree/Tooling/Scripts/Tracking/TrackingManager.cs
using System;

using UnityEngine;

using Ultraleap.TouchFree.Tooling.Configuration;
using Ultraleap.TouchFree.Tooling.Connection;

namespace Ultraleap.TouchFree.Tooling.Tracking
{
    // class: TrackingManager
    // This class provides methods for getting and setting the settings of the tracking software.
    public class TrackingManager
    {
        // Function: RequestTrackingState
        // Used to request a <TrackingState> representing the current state of the tracking software's settings via the
        // WebSocket.
        // Provides a <TrackingState> asynchronously via the _callback parameter.
        //
        // If your _callback requires context it should be bound to that context via .bind()
        public static void RequestTrackingState(Action<TrackingStateResponse> _callback)
        {
            if (_callback == null)
            {
                Debug.LogError("Config file state request failed. This call requires a callback.");
                return;
            }

            ConnectionManager.serviceConnection?.RequestTrackingState(_callback);
        }

        // Function: RequestTrackingChange
        // Requests a modification to the tracking software's settings. Takes any of the following arguments representing
        // the desired changes and sends them through the <ConnectionManager>.
        // <MaskingConfig>, <CameraConfig>, and bools for if images are allowed and if analytics are enabled.
        //
        // Provide a _callback if you require confirmation that your settings were used correctly.
        // If your _callback requires context it should be bound to that context via .bind().
        public static void RequestTrackingChange(
            TrackingState _state,
            Action<TrackingStateResponse> _callback = null)
        {
            Guid requestID = Guid.NewGuid();

            var content = StringifyTrackingState(_state);

            string jsonContent = "";
            jsonContent += "{\"action\":\"";
            jsonContent += ActionCode.SET_TRACKING_STATE + "\",\"content\":{\"requestID\":\"";
            jsonContent += requestID + "\",";
            jsonContent += content + "}}";

            Debug.Log("Sending Tracking Request:");
            Debug.Log(jsonContent);

            if (_callback == null) {
                ConnectionManager.serviceConnection?.RequestTrackingChange(jsonContent, requestID.ToString());
            } else {
                ConnectionManager.serviceConnection?.RequestTrackingChange(jsonContent, requestID.ToString(), _callback);
            }
        }

        private static string StringifyTrackingState(TrackingState _state)
        {
            string newContent = "";

            Debug.Log(JsonUtility.ToJson(_state));

            if (_state.mask.HasValue) {
                newContent += "\"mask\": " + JsonUtility.ToJson(_state.mask.Value) + ",";
            }

            if (_state.allowImages.HasValue) {
                newContent += $"\"allowImages\": {_state.allowImages.Value.ToString().ToLower()},";
            }
            if (_state.cameraReversed.HasValue) {
                newContent += $"\"cameraReversed\": {_state.cameraReversed.Value.ToString().ToLower()},";
            }
            if (_state.analyticsEnabled.HasValue) {
                newContent += $"\"analyticsEnabled\": {_state.analyticsEnabled.Value.ToString().ToLower()},";
            }

            return newContent;
        }
    }
}